tests/foreman/api/test_ansible.py

```diff
@@ -784,3 +784,127 @@ def test_positive_ansible_job_with_nonexisting_module(
         assert [i['output'] for i in result if termination_msg in i['output']]
         assert [i['output'] for i in result if i['output'] == 'StandardError: Job execution failed']
         assert [i['output'] for i in result if i['output'] == 'Exit status: 4']
+
+    @pytest.mark.no_containers
+    @pytest.mark.parametrize('ansible_check_mode', ['True', 'False'], ids=['enabled', 'disabled'])
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    def test_positive_ansible_job_with_check_mode(
+        self,
+        request,
+        target_sat,
+        module_org,
+        module_location,
+        module_ak_with_synced_repo,
+        rhel_contenthost,
+        ansible_check_mode,
+    ):
+        """Verify the Ansible REX job runs with check mode enabled,
+        to see what changes it would make without applying them.
+
+        :id: a082f599-fbf7-4779-aa18-5139e2bce999
+
+        :steps:
+            1. Register a content host with satellite
+            2. Clone the "Ansible Roles - Ansible Default" Job Template.
+            3. Enable "Ansible Check Mode Enabled" on the cloned Job Template.
+            4. Create a custom role to validate ansible check mode, and assign it to the host
+            5. Schedule a Job on host and select the cloned Job Category and Job Template.
+            6. Validate REX job runs in check mode to preview changes without applying them.
+
+        :expectedresults: Ansible REX job works without any errors and,
+            when ansible_check_mode_enabled it previews changes without applying them.
+
+        :verifies: SAT-32223
+        """
+        data = """---
+        - name: Inform if running in check mode
+          debug:
+            msg: "Check mode is enabled. No changes will be applied."
+          when: ansible_check_mode
+
+        - name: Inform if running in normal (non-check) mode
+          debug:
+            msg: "Check mode is not enabled. Changes will be applied."
+          when: not ansible_check_mode
+
+        - name: Attempt to create a test file (only in non-check mode)
+          file:
+            path: /tmp/check_mode_test_file
+            state: touch
+          when: not ansible_check_mode
+
+        - name: Note on file creation task behavior when in check mode
+          debug:
+            msg: "File creation task is conditionally skipped or executed based on check mode."
+          when: ansible_check_mode
```

Comment:
Is this task being used in the test?

Reply:
its not used yet, but it would be good to have for debugging purpose, and file creation task is being validated for `when: not ansible_check_mode`